samples/nextjs/src/components/GraphQL-IntegratedDemo/index.tsx

```diff
@@ -1,7 +1,62 @@
 import { Text, Link } from '@sitecore-jss/sitecore-jss-nextjs';
 import NextLink from 'next/link';
 
-const GraphQLIntegratedDemo = (props) => {
+interface DataSource {
```

Comment:
This way of defining datasource interfaces makes them not reusable.

I propose adding interfaces for all the base field types (SingleLineText, GeneralLink, ettc) to sitecore-jss (first I though they should go into manifest, but this would be needed in Sitecore-first mode too).

Then, each datasource should export de defined as an interface and exported from an independent file. For example, we could create a folder `/src/models` and it would have `GraphQL-IntegratedDemo.model.ts`, which would be something like
```
import { fieldModels } from 'sitecore-jss/models';

export default interface GraphQLIntegratedDemoModel {
  sample1: fieldModels.SingleLineText,
  sample2: fieldModels.GeneralLink,
  ...
}
```

This would allow GraphQLIntegratedDemoModel to be reusable across multiple components, which is more in line with a real implementation.

Also this would be a great step towards contract-first mode. If this is achieved, the next step would be to investigate whether we can generate these datasource models when the manifest runs and template definitions are processed.

Reply:
@anastasiya29 Sorry, I didn't get a point, for example current `sample2` field has type:
```
  sample2: {
    definition: {
      type: string;
      shared: boolean;
    };
    jss: {
      value: {
        href: string;
        linktype: string;
        target: string;
        text: string;
        url: string;
      };
    };
    target: string;
    text: string;
    url: string;
  };
```
and you want simply replace this type by `fieldModels.GeneralLink`. I'm not sure that we can do it, or can you provide example of types: SingleLineText, GeneralLink?
Probably, I'll create a separate PBI for it, it will take more time